docs/book/beyond_basics/webhook_installer_generator.md

```diff
@@ -0,0 +1,148 @@
+```
+
+#### Annotations
+
+Each comment line that starts with `+kubebuilder:webhook:` will be processed to extract annotations.
+
+The annotations can be grouped in 2 categories based on what struct they configure.
+
+The _first_ category is for each individual webhook.
+They are used to set the fields in [`Webhook`](https://github.com/kubernetes-sigs/controller-runtime/blob/a8ea2056444a5d74d7408e4b8798cbe63b14066b/pkg/webhook/admission/webhook.go#L55-L80) struct.
+The annotations for the same webhook are allowed to span across multiple lines as long as they are prefixed with
+`+kubebuilder:webhook:` and in the __same__ comment group.
+
+For example, the following is for one single webhook
+
+```
+// +kubebuilder:webhook:groups=apps,versions=v1,resources=deployments,verbs=create,update
+// +kubebuilder:webhook:name=mutating-create-update-deployment.testproject.org
+// +kubebuilder:webhook:path=/mutating-create-update-deployment
+// +kubebuilder:webhook:type=mutating,failure-policy=fail
+```
+
+`groups`, `versions` and `resources` have the same semantic as the ones used for generating RBAC manifests.
+They can reference a core type or a CRD type.
+
+`verbs` are used to set [`Operations`](https://github.com/kubernetes/api/blob/9fcf73cc980bd64f38a4f721a7371b0ebb72e1ff/admissionregistration/v1beta1/types.go#L234-L243).
+It supports `create`, `update`, `delete`, `connect` and `*` case-insensitively.
+
+`name` is the name of the webhook and
+is used to set [`Name`](https://github.com/kubernetes/api/blob/9fcf73cc980bd64f38a4f721a7371b0ebb72e1ff/admissionregistration/v1beta1/types.go#L146).
+`path` is the endpoint that this webhook serves and
+is used to set [`Path`](https://github.com/kubernetes-sigs/controller-runtime/blob/a8ea2056444a5d74d7408e4b8798cbe63b14066b/pkg/webhook/admission/webhook.go#L61-L62).
+Both `name` and `path` do NOT allow `,` and `;`.
+
+`type` indicates the webhook type which can be either `mutating` or `validating`.
+
+`failure-policy` is used to set [`FailurePolicy`](https://github.com/kubernetes/api/blob/9fcf73cc980bd64f38a4f721a7371b0ebb72e1ff/admissionregistration/v1beta1/types.go#L54-L61).
+It supports `fail` and `ignore` case-insensitively.
+
+The _second_ category is for the webhook server.
+All of them are used to configuration [`ServerOptions`](https://github.com/kubernetes-sigs/controller-runtime/blob/a8ea2056444a5d74d7408e4b8798cbe63b14066b/pkg/webhook/server.go#L39-L98) struct.
+Each annotation should only be used once.
+They don't have to be in the same comment group.
+
+The following is an example using webhook server annotations.
+
+```
+// +kubebuilder:webhook:port=7890,cert-dir=/path/to/cert
+// +kubebuilder:webhook:service=test-system:webhook-service,selector=app:webhook-server
+// +kubebuilder:webhook:secret=test-system:webhook-secret
+// +kubebuilder:webhook:mutating-webhook-config-name=test-mutating-webhook-cfg
+// +kubebuilder:webhook:validating-webhook-config-name=test-validating-webhook-cfg
+```
```

Comment:
same as above. I'd like to see some explanation or an example showing which files these annotations should go into.